src/main/scala/wdl4s/AstTools.scala

```diff
@@ -50,8 +50,8 @@ object AstTools {
               WdlArrayType(member)
             case "Map" =>
               if (subtypes.size != 2) throw new SyntaxError(wdlSyntaxErrorFormatter.mapMustHaveExactlyTwoTypeParameters(typeTerminal))
-              val keyType = subtypes(0).wdlType(wdlSyntaxErrorFormatter)
-              val valueType = subtypes(1).wdlType(wdlSyntaxErrorFormatter)
+              val keyType = subtypes.head.wdlType(wdlSyntaxErrorFormatter)
+              val valueType = subtypes.tail.head.wdlType(wdlSyntaxErrorFormatter)
```

Comment:
Disregarding my typical opinions about compile time safety (since wdl4s is a minefield of these situations), I find the previous more clear as to what's going on. OTOH if this was IntelliJ being ornery then disregard this too